selfservice/strategy/oidc/provider_gitlab.go

```diff
@@ -33,6 +33,36 @@ func NewProviderGitLab(
 	}
 }
 
+func (g *ProviderGitLab) oauth2() (*oauth2.Config, error) {
+
+	endpoint, err := g.endpoint()
+
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	authUrl := *endpoint
+	tokenUrl := *endpoint
+
+	authUrl.Path = path.Join(authUrl.Path, "/oauth/authorize")
+	tokenUrl.Path = path.Join(tokenUrl.Path, "/oauth/token")
```

Comment:
You are converting to string later, maybe just do it right away and don't bother with the URL struct?
```suggestion
	authURL := path.Join(endpoint.String(), "/oauth/authorize")
	tokenURL := path.Join(endpoint.String(), "/oauth/token")
```

Reply:
So this will lead to a missing slash on the scheme. 
e.g. `https:/gitlab.com/oauth/authorize`

Doing it with `fmt.Sprintf` would also be unsafe since someone could do a trailing `issuer_url` e.g. https://example.com/